app/experimenter/experiments/api/v3/serializers.py

```diff
@@ -109,13 +126,36 @@ def validate(self, data):
                         ]
                     }
                 )
+
+        variants = data.get("variants")
+
+        if variants:
+            if not self.is_variant_valid(variants):
```

Comment:
I would name this more descriptively like `all_variant_slugs_unique` or something